### ISSUE TITLE
cors unit test

### ISSUE THREAD

**Issue by austinpapp** [PowerlineApp/powerline-server#278]
@igaponov following up on the cors issue in https://github.com/PowerlineApp/powerline-mobile/issues/205 , let's write a unit test for this please.

**Comment by igaponov** [comment]
This error is occurred when there is a fatal error on backend and php can't send headers in this case, that's why frontend has CORS error. This case is untestable, we can't do anything with this, just check logs to know the real error message.

**Comment by austinpapp** [comment]
not true. however, i should have said functional test and not a unit test. anyway, i may add a quick auxiliary test script to travis (shell or python or something) that will test as such:
```
 17:23 $ curl -H "Origin:*" -I https://api-dev.powerli.ne/
HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Set-Cookie: PHPSESSID=2c2cdnklm79t3fae39vccem3u7; path=/
Cache-Control: no-cache
Date: Sun, 18 Sep 2016 21:19:17 GMT
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: accept, origin, x-requested-with, Authorization, content-type, Token
X-Debug-Token: c5b277
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS, PATCH
```
so, `Access-Control-Allow-Origin: *` proves CORS works

**Comment by igaponov** [comment]
@austinpapp 
Of course there is the header, the code is 200 OK.
Try to get a fatal error, there will be no code, because php can't send headers after fatal error.

If you want to test that there is the header in normal responses, we have a test for it: src/Civix/ApiBundle/Tests/EventListener/CORSSubscriberTest.php

**Comment by austinpapp** [comment]
the header not coming back correctly is what i want to test. if a header isn't coming back, clearly something is wrong. i want to spot that before we release. we should be able to do that during the build. maybe this is something we can do if we build out or use phing correctly as phpunit alone maybe isn't the right spot for this?

**Comment by igaponov** [comment]
@austinpapp the header is coming correctly, the test for it: src/Civix/ApiBundle/Tests/EventListener/CORSSubscriberTest.php.
The header will not come only on fatal error. We can't do anything with it, just need to avoid fatal errors.
